codice/ddf
# [2.19.x] DDF-6013 G-5272 Allow users to remove themselves from the sharing list

#### What does this PR do?
Allows a user who does not want to see a workspace that has been shared with them anymore to be able to remove themselves from the sharing list. 
#### Who is reviewing it? 


#### Select relevant component teams: 


#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining 
@stustison

#### How should this be tested?
- build / unzip
- before starting ddf go into /etc/ directory, update user.attributes file to include
 ```
"test" : {
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress":"test@localhost.local",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role": ["admin","manager","viewer","system-admin","systembundles"]
    }
```
and in user.properties files add 
```
test=test,group,admin,manager,viewer,system-admin,systembundles,ssh
``` 
- login with either admin user or test user with `test` as username/password
- create workspace / result / search forms and share with another user (use test@localhost.local to share with test user), use different access types (read, write, share)
- login user the user you shared the items with 
- verify the `Leave Form/Workspace` button removes user access
- verify with owner of the form that nothing else has changed
- share an item by group and verify that the user cannot leave the item

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #6013 

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
